spec/controllers/tasks_controller_spec.rb

```diff
@@ -391,17 +401,22 @@
         end
 
         let(:test) { build(:test) }
+        let(:new_testing_framework) { create(:testing_framework) }
         let!(:task) { create(:task, valid_attributes.merge(tests: [test])) }
 
-        let(:tests_attributes) { {'0' => test.attributes.merge('title' => 'new_test_title')} }
+        let(:tests_attributes) { {'0' => test.attributes.merge('title' => 'new_test_title', 'testing_framework_id' => new_testing_framework.id)} }
```

Comment:
Is there any reason you are using the old rocket-styled syntax for the has? What about using `testing_framework_id: new_testing_framework.id` (and similar for the others)?

Reply:
Originally I just wanted the syntax to be consistent with the already existing code but after your comment I tried to use the ':' syntax instead and found that the tests actually fail when you use 
```'title': 'new_test_title'``` instead of ```'title' => 'new_test_title'```.
You can see why when you output test.attributes.merge('title' => 'new_test_title') in the console:
```
{"id"=>1, "created_at"=>..., "title"=>"title", "description"=>..., :title=>"new_test_title"}
```
So apparently merge() converts strings to symbols when you use the ':' syntax but not when you use the rocket syntax. The same happens with double-quoted strings as keys.
I decided to work around that with symbolize_keys and I also changed the rest of the context to use the new hash syntax.

Reply:
Ah, that sounds plausible. But thanks for changing, I think it looks cleaner (even though the additional `symbolize_keys` call is required).